modules/nextflow/src/test/groovy/nextflow/k8s/K8sConfigTest.groovy

```diff
@@ -140,6 +141,29 @@ class K8sConfigTest extends Specification {
         client.server == 'http://foo'
         client.namespace == 'this'
         client.serviceAccount == 'that'
+        client.httpConnectionTimeout == null // testing default
+        client.httpReadTimeout == null // testing default
+
+    }
+
+    def 'should create client config with http request timeouts' () {
+
+        given:
+        def CONFIG = [
+                namespace: 'this',
+                serviceAccount: 'that',
+                client: [server: 'http://foo', httpReadTimeout: '20s', httpConnectionTimeout: '25s']
```

Comment:
I think it would be better to keep those settings in the top-most scope (like `namespace`) otherwise they will be ignored when the client is built via the (kube) context. See here 

https://github.com/nextflow-io/nextflow/blob/master/modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy#L210-L221